core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3626,31 +3645,17 @@ private void startSyncAndJournal(RpcContext rpcContext,
       throws InvalidPathException, IOException, ConnectionFailedException {
     try (LockResource r = new LockResource(mSyncManager.getSyncManagerLock())) {
       mSyncManager.startSync(lockedInodePath.getUri());
+      MountTable.Resolution resolution = mMountTable.resolve(lockedInodePath.getUri());
+      long mountId = resolution.getMountId();
       AddSyncPointEntry addSyncPoint =
           AddSyncPointEntry.newBuilder()
-              .setSyncpointPath(lockedInodePath.getUri().toString()).build();
+              .setSyncpointPath(lockedInodePath.getUri().toString())
+              .setMountId(mountId)
+              .build();
       rpcContext.journal(JournalEntry.newBuilder().setAddSyncPoint(addSyncPoint).build());
```

Comment:
This journaling isn't very reliable - what if we add the sync point and then hit a runtime exception? The sync point will be added but we won't journal. We need a better story for how we keep master metadata in sync with what's written to the journal. 

In the inode tree, we achieve this by only allowing metadata to be modified via `applyAndJournal(JournalEntry)` methods which apply updates using a `JournalEntry`, then log the `JournalEntry`. If the journal cannot be applied, the master dies, so we keep the `apply` section as basic as possible. See `InodeTreePersistentState`